### PR TITLE
test(world-cli): Add a sample test and fix a bug with the docker daemon check

### PR DESCRIPTION
**Pull request by jerargus**
## What is the purpose of the change
  
Add a sample test to show how to capture the standard out and standard error for tests.

## Testing and Verifying
  
run `go test ./...` from the root directory to see this test in action.

### CODE REVIEW

**Comment by smsunarto** [comment]
@jerargus while we are this, i think we should add a go and/or docker dependency to the world cardinal subcommand. These subcommand doesn't use bubbletea, so we would should extract https://github.com/Argus-Labs/world-cli/blob/main/common/tea_cmd/dependencies.go#L64-L78 into a function we can directly call without tea.Cmd/Msg wrapper

**Comment by jerargus** [comment]
> @jerargus while we are at this, i think we should add a go and/or docker dependency to the world cardinal subcommand. These subcommand doesn't use bubbletea, so we would should extract https://github.com/Argus-Labs/world-cli/blob/main/common/tea_cmd/dependencies.go#L64-L78 into a function we can directly call without tea.Cmd/Msg wrapper

File: cmd/world/cardinal/create.go [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+	"pkg.world.dev/world-cli/common/dependency"
 	"pkg.world.dev/world-cli/common/tea_cmd"
 	"pkg.world.dev/world-cli/tea/component/steps"
 	"pkg.world.dev/world-cli/tea/style"
-	"strings"
 )
 
 const TemplateGitUrl = "https://github.com/Argus-Labs/starter-game-template.git"
 
-var CreateDeps = []tea_cmd.Dependency{
```
Comment (smsunarto) [inline cmd/world/cardinal/create.go:15]: wouldn't this be run twiced if we also use `PersistentPreRunE`?

**Review by smsunarto — APPROVED** [review]

File: cmd/world/cardinal/create.go (hunk 1 above)
Comment (jerargus) [reply cmd/world/cardinal/create.go:15]: Right. I've removed the extra dependency check.

**Pull request merged**